doc/lsst.obs.base/creating-an-obs-package.rst

```diff
@@ -0,0 +1,28 @@
+.. creating-an-obs-package:
+
+####################################
+How to create a new LSST obs package
+####################################
+
+These instructions describe how to create a package that allows the LSST Data Management Science Pipeline to read raw data from a new telescope and instrument.
+
+1. MetadataTranslator
+=====================
+
+Create a ``MetadataTranslator`` derived from `~astro_metadata_translator.translators.MetadataTranslator` or `~astro_metadata_translator.translators.FitsTranslator`.
+
+2. Formatter
+============
+
+Create a ``FitsRawFormatter`` derived from `~lsst.obs.base.FitsRawFormatterBase`.
+
+At minimum, override the abstract methods `~lsst.obs.base.FitsRawFormatterBase.makeVisitInfo` and `~lsst.obs.base.FitsRawFormatterBase.getDetector`.
+
+3. Instrument
+=============
+
+Create an ``Instrument`` derived from `~lsst.daf.butler.Instrument` (NOTE: this base class and its respective test framework is planned to be moved to `lsst.obs.base`).
+
+First write a test case, derived from `~lsst.daf.butler.instrument_tests.InstrumentTests` and `lsst.utils.tests.TestCase`, defining ``self.data`` and ``self.instrument`` in ``setUp``. Run this test via ``pytest -sv tests/test_newObs.py``: all of the tests should fail.
+
+Next, at minimum, override these abstract methods: `~lsst.daf.butler.Instrument.getName`, `~lsst.daf.butler.Instrument.getCamera`.
```

Comment:
I'm not sure we want to make `getCamera` part of the `Instrument` base class API, as it implies that there is a nominal camera that can be retrieved without providing a calibration repository.  That's true right now for all cameras, of course, because it's all Gen2 supports.  But I'm not sure it will be true forever.

That said, I'm not sure it *won't* be true forever, and maybe refactoring all of the cameras if/when we make that change isn't a big deal.  But that's the reason why `getCamera` was only defined on `HyperSuprimeCam`, not `Instrument`, before.

Reply:
Yup, I knew that context going in. The problem is that we have to actually choose whether to make `getCamera` part of the `Instrument` API or not at all. Right now, we need it in order to have `FitsRawFormatter.getDetector()` work (which we need for the new SkyWcs code), so I've made it a proper part of the API, so that `getDetector` works in the other Cameras.

Do you have a different approach you would recommend to deal with this until versioned calibration repos become available?

Reply:
I do wonder if we have some guidance as to what `getName` should return. `ObservationInfo` defines an instrument name and a telescope name. How should this name relate to the translator name?

Reply:
Good question: in the tests I've written on this ticket, I'm assuming that `getName()==camera.getName()`, which should probably also equal `observationInfo.instrument_name`.